sim/src/NA6PVerTel.cxx

```diff
@@ -107,23 +119,31 @@ void NA6PVerTel::createGeometry(TGeoVolume* world)
   TGeoVolume* pixelSensor = new TGeoVolume("PixelSensor", sensorShape, NA6PTGeoHelper::instance().getMedium(addName("Silicon")));
   pixelSensor->SetLineColor(NA6PTGeoHelper::instance().getMediumColor(addName("Silicon")));
+  auto* carbonplateShape = new TGeoBBox("CarbonPlateShape", carbonPlateDX / 2, carbonPlateDY / 2, carbonPlateDz / 2);
+  TGeoVolume* cbPlate = new TGeoVolume("CarbonPlate", carbonplateShape, NA6PTGeoHelper::instance().getMedium(addName("CarbonFiber")));
+  cbPlate->SetLineColor(NA6PTGeoHelper::instance().getMediumColor(addName("CarbonFiber")));
 
   // place sensors to station
   std::vector<float> alpdx{pixChipDX / 2 + pixChipOffsX, -pixChipDX / 2 + pixChipOffsY, -pixChipDX / 2 - pixChipOffsX, pixChipDX / 2 - pixChipOffsY};
   std::vector<float> alpdy{pixChipDY / 2 - pixChipOffsY, pixChipDY / 2 + pixChipOffsX, -pixChipDY / 2 + pixChipOffsY, -pixChipDY / 2 - pixChipOffsX};
   for (size_t ii = 0; ii < alpdx.size(); ++ii) {
     auto* sensorTransform = new TGeoTranslation(alpdx[ii], alpdy[ii], 0);
     pixelStationVol->AddNode(pixelSensor, composeSensorVolID(ii), sensorTransform);
+    int nns = pixelStationVol->GetNodes()->GetEntries() - 1;
+    auto* cbTransform = new TGeoTranslation(alpdx[ii], alpdy[ii], pixChipDz / 2 + carbonPlateDz / 2);
+    pixelStationVol->AddNode(cbPlate, composeNonSensorVolID(ii + 40), cbTransform);
+    int nnc = pixelStationVol->GetNodes()->GetEntries() - 1;
+    LOGP(info, "Adding nodes {} and {} to {}, n. nodes = {}", pixelStationVol->GetNode(nns)->GetName(), pixelStationVol->GetNode(nnc)->GetName(), pixelStationVol->GetName(), pixelStationVol->GetNodes()->GetEntries());
   }
   // place frames + sensor stations
   float zoffs = param.posVerTelPlaneZ[0] + boxDZ / 2 - boxDZMargin; // offset to be added due to the placement of stations to the VT box
   for (int ll = 0; ll < param.nVerTelPlanes; ++ll) {
     auto* stationTransform = new TGeoCombiTrans(param.posVerTelPlaneX[ll], param.posVerTelPlaneY[ll], param.posVerTelPlaneZ[ll] - zoffs,
                                                 NA6PTGeoHelper::rotAroundVector(0.0, 0.0, 0.0, 0.0));
     vtContainer->AddNode(pixelStationVol, composeNonSensorVolID(ll), stationTransform);
-    auto* frameTransform = new TGeoCombiTrans(param.posVerTelPlaneX[ll], param.posVerTelPlaneY[ll], param.posVerTelPlaneZ[ll] + 0.5 * (frameDZ + pixChipDz) - zoffs,
-                                              NA6PTGeoHelper::rotAroundVector(0, 0.0, 0.0, 0.0));
-    vtContainer->AddNode(pixStFrame, composeNonSensorVolID(ll + 20), frameTransform);
+    //    auto* frameTransform = new TGeoCombiTrans(param.posVerTelPlaneX[ll], param.posVerTelPlaneY[ll], param.posVerTelPlaneZ[ll] + 0.5 * (frameDZ + pixChipDz) - zoffs,
+    //                                              NA6PTGeoHelper::rotAroundVector(0, 0.0, 0.0, 0.0));
+    //    vtContainer->AddNode(pixStFrame, composeNonSensorVolID(ll + 20), frameTransform);
```

Comment:
The old pixStFrame volume is created above but it is never added. If it is not needed (see the comment above about the plate) then it should be removed from the code.